api/src/main/java/com/epam/pipeline/controller/plugin/PluginController.java

```diff
@@ -0,0 +1,138 @@
+    @GetMapping
+    @ApiOperation(
+            value = "List UI plugins",
+            notes = "Returns a list of UI plugins, optionally filtered by type.",
+            response = UIPlugin.class,
+            responseContainer = "List")
+    @ApiResponses({@ApiResponse(code = HTTP_STATUS_OK, message = "Successfully retrieved plugins")})
+    public ResponseEntity<List<UIPlugin>> getPlugins(
+            @ApiParam(value = "Optional plugin type filter (LaunchForm or RunLog)")
+            @RequestParam(required = false) PluginType type) {
+        return ResponseEntity.ok(pluginService.getPlugins(type));
+    }
+
+    @PostMapping
+    @ApiOperation(
+            value = "Create or update a UI plugin",
+            notes = "Creates a new plugin if ID is absent, updates existing plugin if ID is provided.",
+            response = UIPlugin.class)
+    @ApiResponses({@ApiResponse(code = HTTP_STATUS_OK, message = "Plugin created or updated successfully")})
+    public ResponseEntity<UIPlugin> savePlugin(
+            @ApiParam(value = "Plugin details", required = true)
+            @RequestBody UIPlugin plugin) {
+        return ResponseEntity.ok(pluginService.savePlugin(plugin));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(
+            value = "Get a UI plugin by ID",
+            notes = "Returns the plugin with the specified ID.",
+            response = UIPlugin.class)
+    @ApiResponses({@ApiResponse(code = HTTP_STATUS_OK, message = "Plugin retrieved successfully")})
+    public ResponseEntity<UIPlugin> getPlugin(
+            @ApiParam(value = "Plugin ID", required = true)
+            @PathVariable Long id) {
+        return ResponseEntity.ok(pluginService.getPlugin(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation(
+            value = "Delete a UI plugin",
+            notes = "Deletes the plugin with the specified ID.")
+    @ApiResponses({@ApiResponse(code = HTTP_STATUS_OK, message = "Plugin deleted successfully")})
+    public void deletePlugin(
+            @ApiParam(value = "Plugin ID", required = true)
+            @PathVariable Long id) {
+        pluginService.deletePlugin(id);
+    }
+
+    @GetMapping(
+            value = "/{id}/content",
+            produces = {"text/javascript", "application/javascript", "text/css"})
+    @ApiOperation(
+            value = "Get plugin file content",
+            notes = "Returns the content of a plugin file specified by ID and relative path.",
+            response = String.class)
+    @ApiResponses({@ApiResponse(code = HTTP_STATUS_OK, message = "File content retrieved successfully")})
+    public ResponseEntity<byte[]> getPluginFileContent(
+            @ApiParam(value = "Plugin ID", required = true)
+            @PathVariable Long id,
+            @ApiParam(value = "Relative file path within plugin directory", required = true)
+            @RequestParam String path) {
+        return ResponseEntity.ok(pluginService.getPluginFileContent(id, path));
```

Comment:
Let's use for this method `writeStreamToResponse` method from `AbstractRestController` class